pkg/lib/cockpit-components-file-autocomplete.jsx

```diff
@@ -113,78 +112,67 @@ export class FileAutoComplete extends React.Component {
         });
     }
 
-    onValueChanged(value) {
-        if (value.length == 0)
-            return;
-
-        if (this.props.onChange)
-            this.props.onChange(value[0].path);
-
-        this.setState({ value: value[0].path });
-
-        if (value[0].type == 'directory') {
-            this.setState({ directory: value[0].path });
-            this.updateFiles(value[0].path);
+    onFilter(event) {
+        if (event.target.value == "" || event.target.value.slice(-1) == "/") {
+            this.setState({ directory: event.target.value || "/" });
+            this.updateFiles(event.target.value || "/");
         }
-    }
 
-    onInputChanged(value) {
-        if (this.props.onChange)
-            this.props.onChange(value);
+        const res = event.target.value !== '' ? this.state.displayFiles.filter(file => file.path.startsWith(event.target.value)) : this.state.displayFiles;
+        return res.map(option => (
+            <SelectOption key={option.path}
+                          className={option.type}
+                          value={{
+                              ...option,
+                              toString: function() { return this.path },
```

Comment:
This looks odd -- doesn't function() introduce a new `this`? I.e. shouldn't this be `toString: () => this.path`?

There are also some more toString() methods in this PR with the same issue.

Reply:
That's exactly what I want - the new `this` from the context of the class that this function is going to be used at at the side of the @patternfly/react-core code.

Reply:
So this is here on the purpose - the arrow notation would break the functionality.

Reply:
Ack, thanks!